runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java

```diff
@@ -384,6 +389,7 @@ private GroundItem buildGroundItem(final Tile tile, final TileItem item)
 		final int realItemId = itemComposition.getNote() != -1 ? itemComposition.getLinkedNoteId() : itemId;
 		final int alchPrice = itemComposition.getHaPrice();
 		final boolean dropped = tile.getWorldLocation().equals(client.getLocalPlayer().getWorldLocation()) && droppedItemQueue.remove(itemId);
+		final boolean onTable = tableItemQueue.remove(itemId);
```

Comment:
I am fairly certain that `tile.getItemLayer().getHeight() > 0` is true iif the item is on a table, so maybe that can just be used instead.